### PR TITLE
fix: fix x/foundation EndBlocker

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

This patch will fix `EndBlocker()` of `x/foundation` and add the corresponding unit tests.

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [x] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Review by dudong2 — APPROVED** [review]
LGTM

File: x/foundation/keeper/keys.go [hunk 1]
```diff
@@ -70,39 +79,38 @@ func voteKey(proposalID uint64, voter sdk.AccAddress) []byte {
...
-// 	begin := len(proposalByVPEndKeyPrefix) + 1
-// 	end := begin + int(key[begin-1]) // uint64
-// 	proposalID = Uint64FromBytes(key[begin:end])
+func splitProposalByVPEndKey(key []byte) (vpEnd time.Time, id uint64) {
+	prefix := proposalByVPEndKeyPrefix
+	begin := len(prefix)
+	end := begin + lenTime
+	vpEnd, err := sdk.ParseTimeBytes(key[begin:end])
+	if err != nil {
+		panic(err)
```
Comment (zemyblue) [inline x/foundation/keeper/keys.go:106]: Isn't there this panic case?
There is not unittest.

Comment (0Tech) [reply x/foundation/keeper/keys.go:106]: You can add a unit test that covers this line. But in the sane condition, it won't happen. There are many similar lines, and of course, x/foundation is not only the case. I think it won't be necessary to add the unit test.

Comment (zemyblue) [reply x/foundation/keeper/keys.go:106]: OK, I understand.

**Review by zemyblue — APPROVED** [review]

**Pull request merged**